### PR TITLE
Gulp Build Script

### PR DESCRIPTION
**Pull request by chadfawcett**
Build process that adds template to angular template cache so module can be included into any project without having route issues with the template files.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  return gulp.src('angular-date-pickable.html')
+    .pipe(templateCache('temp-templates.js', { module: 'jbDatePickable' }))
+    .pipe(gulp.dest('build'));
+});
+
+gulp.task('clean', ['templates', 'dist'], function() {
+  return del(['build']);
+});
+
+gulp.task('watch', ['build'], function() {
```
Comment (jonotron) [inline gulpfile.js]: Please remove this watch task. `gulp.watch` doesn't work with newly added files, and I'd rather the build tasks not provide a watch capability to minimize the dependencies and opinions. Rather, developers should use their own watch tool like https://www.npmjs.com/package/watch or similar to their liking.

**Comment by jonotron** [comment]
@chadfawcett Looks great! Just the removal of the `watch` task and it's :+1:

**Comment by chadfawcett** [comment]
@jonotron Good to go!

**Comment by jonotron** [comment]
Thanks for the changes!

**Pull request merged**